src/convert/base.jl

```diff
@@ -296,8 +298,10 @@ end
 function sexp(::Type{RClass{:list}}, a::AbstractArray)
     ra = protect(allocArray(VecSxp, size(a)...))
     try
-        for i in 1:length(a)
-            ra[i] = a[i]
+        # we want this to work even if a doesn't use one-based indexing
+        # we only care about ra having the same length (which it does)
+        for (i, idx) in zip(eachindex(ra), eachindex(a))
+            ra[i] = a[idx]
```

Comment:
This feels like a situation where a `copyto!` method could make sense, assuming such a method doesn't already exist (I assume it doesn't)

Reply:
Hmmm, yeah, but I'm going to punt on that until I've tackled #444